### PR TITLE
ref(api): move participant join and left to middleware

### PR DESCRIPTION
**Pull request by virtuacoplenny**

### CODE REVIEW

File: react/features/display-name/functions.js [hunk 1]
```diff
@@ -7,7 +7,7 @@
  * @param {string} suffix - Suffix that will be appended.
  * @returns {string} The formatted display name.
  */
-export function appendSuffix(displayName: string, suffix: string) {
+export function appendSuffix(displayName: string, suffix: ?string) {
```
Comment (saghul) [inline react/features/display-name/functions.js]: wouldn't it be better to give it a default value of `''` ?

Comment (virtuacoplenny) [reply react/features/display-name/functions.js]: Yes. That would reduce the perceived variable arity, which I feel would be simpler to read.

File: react/features/display-name/functions.js [hunk 2]
```diff
@@ -7,7 +7,7 @@
  * @param {string} suffix - Suffix that will be appended.
  * @returns {string} The formatted display name.
  */
-export function appendSuffix(displayName: string, suffix: ?string) {
+export function appendSuffix(displayName: string, suffix: string = '') {
```
Comment (saghul) [inline react/features/display-name/functions.js]: then I guess we can skip the check for suffix's truthyness?

**Comment by saghul** [comment]
Can you pleasee rebase?

**Pull request merged**